spec/intl.html

```diff
@@ -0,0 +1,372 @@
+        1. If _matcher_ is `"basic"`, then
+          1. Let _bestFormat_ be BasicFormatMatcher(_opt_, _formats_).
+        1. Else,
+          1. Let _bestFormat_ be BestFitFormatMatcher(_opt_, _formats_).
+        1. For each row in <emu-xref href="#table-datetimeformat-components"></emu-xref>, except the header row, in table order, do
+          1. Let _prop_ be the name given in the Property column of the row.
+          1. Let _p_ be _bestFormat_.[[&lt;_prop_&gt;]].
+          1. If _p_ not *undefined*, then
+            1. Set _dateTimeFormat_'s internal slot whose name is the Internal Slot column of the row to _p_.
+        1. If _dateTimeFormat_.[[Hour]] is not *undefined*, then
+          1. Let _hcDefault_ be _dataLocaleData_.[[hourCycle]].
+          1. Let _hc_ be _dateTimeFormat_.[[HourCycle]].
+          1. If _hc_ is *null*, then
+            1. Set _hc_ to _hcDefault_.
+          1. If _hour12_ is not *undefined*, then
+            1. If _hour12_ is *true*, then
+              1. If _hcDefault_ is `"h11"` or `"h23"`, then
+                1. Set _hc_ to `"h11"`.
+              1. Else,
+                1. Set _hc_ to `"h12"`.
+            1. Else,
+              1. Assert: _hour12_ is *false*.
+              1. If _hcDefault_ is `"h11"` or `"h23"`, then
+                1. Set _hc_ to `"h23"`.
+              1. Else,
+                1. Set _hc_ to `"h24"`.
+          1. Set _dateTimeFormat_.[[HourCycle]] to _hc_.
+          1. If _dateTimeformat_.[[HourCycle]] is `"h11"` or `"h12"`, then
+            1. Let _pattern_ be _bestFormat_.[[pattern12]].
+          1. Else,
+            1. Let _pattern_ be _bestFormat_.[[pattern]].
+        1. Else,
+          1. Set _dateTimeFormat_.[[HourCycle]] to *undefined*.
+          1. Let _pattern_ be _bestFormat_.[[pattern]].
+        1. Set _dateTimeFormat_.[[Pattern]] to _pattern_.
+        1. <ins>For each row in <emu-xref href="#table-temporal-patterns"></emu-xref>, except the header row, in table order, do</ins>
+          1. <ins>Let _limitedOptions_ be a new Record.</ins>
+          1. <ins>Let _fields_ be the list of fields in the Fields column of the row.</ins>
+          1. <ins>For each field _field_ of _opts_, do</ins>
+            1. <ins>If _field_ is in _fields_, then</ins>
+              1. <ins>Set _limitedOptions_.[[&lt;field>]] to _opts_.[[&lt;field>]].</ins>
+          1. <ins>If _matcher_ is `"basic"`, then</ins>
+            1. <ins>Let _bestFormat_ be BasicFormatMatcher(_limitedOptions_, _formats_).</ins>
+          1. <ins>Else,</ins>
+            1. <ins>Let _bestFormat_ be BestFitFormatMatcher(_limitedOptions_, _formats_).</ins>
+          1. <ins>If _bestFormat_ has any fields that are in _fields_, then</ins>
+            1. <ins>If _bestFormat_ has an [[hour]] field, then</ins>
+              1. <ins>Do the hourCycle dance.</ins>
```

Comment:
If you make a new abstract operation, then you don't have to duplicate complex parts of the spec like the "hourCycle dance".